src/org/labkey/test/tests/DomainFieldTypeChangeTest.java

```diff
@@ -0,0 +1,205 @@
+        clickButton("Next");
+
+        setFormElement(Locator.name("name"), runName);
+        setFormElement(Locator.name("runTestInteger"), "12");
+        setFormElement(Locator.name("runTestDecimal"), "1.12");
+        setFormElement(Locator.name("runTestDate"), "01-03-2022");
+        checkRadioButton(Locator.radioButtonById("Fileupload"));
+        setFormElement(Locator.input("__primaryFile__"), runFile);
+        clickButton("Save and Finish");
+
+        goToManageAssays();
+        clickAndWait(Locator.linkWithText(assayName));
+        waitForElement(Locator.linkWithText(runName));
+        click(Locator.linkWithText("Manage assay design"));
+        assayDesignerPage = _assayHelper.clickEditAssayDesign();
+        runFields = assayDesignerPage.goToRunFields();
+        runFields.getField("runTestInteger").setType(FieldDefinition.ColumnType.Decimal);
+        checker().verifyEquals("Incorrect error message for integer to decimal", Arrays.asList("Cannot convert from INTEGER to DOUBLE for non-provisioned table"), assayDesignerPage.clickSaveExpectingErrors());
+        runFields.getField("runTestInteger").setType(FieldDefinition.ColumnType.String);
+        runFields.getField("runTestBoolean").setNumberFormat("yes;no");
+
+        batchFields = assayDesignerPage.goToBatchFields();
+        batchFields.getField("batchTestDate").setType(FieldDefinition.ColumnType.String);
+        batchFields.getField("batchTestDecimal").setType(FieldDefinition.ColumnType.String);
+        assayDesignerPage.clickFinish();
+
+//        assayDesignerPage = _assayHelper.clickEditAssayDesign();
+//        runFields = assayDesignerPage.goToRunFields();
+//        runFields.getField("runTestBoolean").setType(FieldDefinition.ColumnType.String);
+//        assayDesignerPage.clickFinish();
+
+        goToManageAssays();
+        clickAndWait(Locator.linkWithText(assayName));
+        table = new DataRegionTable("Runs", getDriver());
+        checker().verifyEquals("Run fields : Incorrect value after changing Integer to string", Arrays.asList("12"),
+                table.getColumnDataAsText("runTestInteger"));
+        checker().verifyEquals("Run fields : Incorrect value after changing Boolean to string", Arrays.asList("no"),
+                table.getColumnDataAsText("runTestBoolean"));
+        checker().verifyEquals("Batch fields : Incorrect value after changing Date to string", Arrays.asList("2022-01-01 00:00:00"),
+                table.getColumnDataAsText("Batch/batchTestDate"));
+        checker().verifyEquals("Batch fields : Incorrect value after changing Decimal to string", Arrays.asList("1.1"),
+                table.getColumnDataAsText("Batch/batchTestDecimal"));
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+    }
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return null;
```

Comment:
Is this correct? Should we return anything here?

Reply:
Thats a legacy method, should be present.